docs/Tiered-Storage-on-Tachyon.md

```diff
@@ -1,90 +1,208 @@
-    -Dtachyon.worker.tieredstore.level1.dirs.path=/mnt/ssd1,/mnt/ssd2
-    -Dtachyon.worker.tieredstore.level1.dirs.quota=60GB,80GB
-    -Dtachyon.worker.allocate.strategy.class=tachyon.worker.block.allocator.MaxFreeAllocator
-    -Dtachyon.worker.evict.strategy=LRU
-
-In this example, there are two storage layers. The alias of the first layer is MEM with one
-directory. The path is /mnt/ramdisk with 10GB quota. The alias of the second layer is SSD with two
-directories. The paths are /mnt/ssd1 and /mnt/ssd2, with 60GB and 80GB quotas respectively. The
-space allocation strategy is MAX_FREE and the block eviction strategy is LRU.
+Tachyon supports tiered storage, which allows Tachyon to manage other storage types besides memory.
+Currently, Tachyon Tiered Storage supports these storage types or tiers:
+
+* MEM (Memory)
+* SSD (Solid State Drives)
+* HDD (Hard Drives)
+
+Using Tachyon with tiered storage allows Tachyon to manage more space in the Tachyon namespace,
+since MEM capacity may still be limited in some deployments. With tiered storage, Tachyon 
+automatically manages blocks between all the configured tiers, so users and administrators do not
+have to manually manage the locations of the data.
+
+# Using Tiered Storage
+
+With the introduction of tiers, the data blocks managed by Tachyon are not necessarily in memory;
+blocks can be in any of the available tiers. To manage the placement and movement of the blocks,
+Tachyon uses *allocators* and *evictors* to place and re-arrange blocks between the tiers. In
+Tachyon, the tiers are ordered from fastest to slowest, from top to bottom. Therefore, the typical
+tiered storage configuration defines the top tier to be MEM, followed by SSD, and finally HDD.
+
+## Storage Directories
+
+A tier is made up of at least 1 storage directory. This directory is a file path where the Tachyon
+blocks should be stored. Tachyon supports configuring multiple directories for a single tier,
+allowing multiple mount points or storage devices for a particular tier. For example, if you have
+five SSD devices on your Tachyon worker, you can configure Tachyon to use all five devices for the
+SSD tier. Configuration for this is described below. Choosing which directory the data should placed
+is determined by the allocator, described later.
+
+## Writing Data
+
+When a user writes a new block, it is written to the MEM tier by default (a custom allocator can be
+used if the default behavior is not desired). If there is not enough space for the block in the MEM
+tier, then the evictor is triggered in order to free space for the new block.
+
+## Reading Data
+
+Reading a data block with tiered storage is similar to standard Tachyon. Tachyon will simply read
+the block from where it is already stored. If Tachyon is configured with multiple tiers, then the
+block will not be necessarily read from the MEM tier, since it could have been moved to a lower tier
+transparently.
+
+### Promoting Blocks
```

Comment:
Let's make logical changes to in another PR? This is just to merge the work in 0.7 back, but the compatibility changes can be made here as kind of a merge conflict resolution.